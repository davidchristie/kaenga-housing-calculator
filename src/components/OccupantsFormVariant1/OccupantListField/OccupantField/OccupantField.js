import React, { Component, PropTypes } from 'react'
import { Button, ListGroupItem } from 'react-bootstrap'
import { Field } from 'redux-form'

import AgeField from './AgeField'

class OccupantField extends Component {
  render () {
    const { index, occupant, remove } = this.props
    return (
      <ListGroupItem>
        <h4>Occupant #{index + 1}</h4>
        <Field
          name={`${occupant}.age`}
          component={AgeField}
        />
        <Button onClick={remove}>
          Remove Occupant
        </Button>
      </ListGroupItem>
    )
  }
}

OccupantField.defaultProps = {
  index: 0,
  occupant: ''
}

OccupantField.propTypes = {
  index: PropTypes.number.isRequired,
  occupant: PropTypes.string.isRequired,
  remove: PropTypes.func
}

export default OccupantField
