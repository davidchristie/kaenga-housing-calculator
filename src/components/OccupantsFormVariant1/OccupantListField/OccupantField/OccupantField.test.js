import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import OccupantField from './OccupantField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<OccupantField />)).toMatchSnapshot()
})
