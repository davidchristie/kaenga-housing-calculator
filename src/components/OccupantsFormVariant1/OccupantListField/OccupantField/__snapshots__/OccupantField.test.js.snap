exports[`test matches snapshot 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  listItem={false}>
  <h4>
    Occupant #
    1
  </h4>
  <Field
    component={[Function]}
    name=".age" />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={undefined}>
    Remove Occupant
  </Button>
</ListGroupItem>
`;
